server/resp/src/main/java/org/infinispan/server/resp/commands/list/RPUSH.java

```diff
@@ -37,19 +37,18 @@ public CompletionStage<RespRequestHandler> perform(Resp3Handler handler,
          return handler.myStage();
       }
 
-      byte[] key = arguments.get(0);
-      // TODO: putAll operation on multimap
+      // TODO: putAll operation on multimap ?
       return pushAndReturn(handler, ctx, arguments);
    }
 
    protected CompletionStage<RespRequestHandler> pushAndReturn(Resp3Handler handler,
                                                                ChannelHandlerContext ctx,
                                                                List<byte[]> arguments) {
       byte[] key = arguments.get(0);
-      MultimapCache<byte[], byte[]> multimap = handler.getMultimap();
+      EmbeddedMultimapListCache<byte[], byte[]> listMultimap = handler.getListMultimap();
       AggregateCompletionStage<Void> aggregateCompletionStage = CompletionStages.aggregateCompletionStage();
       for (int i = 1; i < arguments.size(); i++) {
-         aggregateCompletionStage.dependsOn(multimap.put(key, arguments.get(i)));
+         aggregateCompletionStage.dependsOn(listMultimap.offerFirst(key, arguments.get(i)));
```

Comment:
The final list will have the elements in the inverse order.

Reply:
true, I used the wrong method